gateway/delete_api_cache.go

```diff
@@ -0,0 +1,12 @@
+package gateway
+
+import (
+	"github.com/TykTechnologies/tyk/storage"
+)
+
+func (gw *Gateway) invalidateAPICache(apiID string) bool {
+	keyPrefix := "cache-" + apiID
+	matchPattern := keyPrefix + "*"
+	store := storage.RedisCluster{KeyPrefix: keyPrefix, IsCache: true, ConnectionHandler: gw.StorageConnectionHandler}
```

Comment:
The connection won't be reused for each deletion:

- suggestion: `invalidateApiCache(store storage.Storage, apiID string) bool`
- we're not keeping the various RedisCluster instances around? Could have sworn there's fields in *Gateway per-storage?
- I don't think RedisCluster.KeyPrefix needs filling, it's part of matchPattern